integration/spark/src/main/scala/org/apache/carbondata/spark/rdd/CarbonDataLoadRDD.scala

```diff
@@ -462,3 +478,133 @@ class CarbonDataLoadRDD[K, V](
   }
 }
 
+class CarbonRDDDataLoadRDD[K, V](
+                               sc: SparkContext,
+                               result: DataLoadResult[K, V],
+                               carbonLoadModel: CarbonLoadModel,
+                               var storeLocation: String,
+                               hdfsStoreLocation: String,
+                               kettleHomePath: String,
+                               columinar: Boolean,
+                               loadCount: Integer,
+                               tableCreationTime: Long,
+                               schemaLastUpdatedTime: Long,
+                               prev: RDD[Row])
+  extends RDD[(K, V)](prev)
+    with Logging {
+
+  sc.setLocalProperty("spark.scheduler.pool", "DDL")
+
+  @DeveloperApi
+  override def compute(theSplit: Partition, context: TaskContext): Iterator[(K, V)] = {
+    val LOGGER = LogServiceFactory.getLogService(this.getClass.getName)
+    val resultIter = new Iterator[(K, V)] {
+      var partitionID = "0"
+      val loadMetadataDetails = new LoadMetadataDetails()
+      var uniqueLoadStatusId = carbonLoadModel.getTableName + CarbonCommonConstants.UNDERSCORE +
+        theSplit.index
+      try {
+        loadMetadataDetails.setPartitionCount(partitionID)
+        loadMetadataDetails.setLoadStatus(CarbonCommonConstants.STORE_LOADSTATUS_FAILURE)
+        carbonLoadModel.setPartitionId(partitionID)
+        carbonLoadModel.setSegmentId(String.valueOf(loadCount))
+        carbonLoadModel.setTaskNo(String.valueOf(theSplit.index))
+
+        storeLocation = CarbonDataLoadRDD.initialize(carbonLoadModel, theSplit.index)
+        loadMetadataDetails.setLoadStatus(CarbonCommonConstants.STORE_LOADSTATUS_SUCCESS)
+        val rddIteratorKey = UUID.randomUUID().toString
+        try{
+          RddInputUtils.put(rddIteratorKey,
+            new RddIterator(firstParent[Row].iterator(theSplit, context), carbonLoadModel))
+          carbonLoadModel.setRddIteratorKey(rddIteratorKey)
```

Comment:
Why can't you just set the iterator object itself in the carbonLoadModel instead of putting it in a global map?